### PR TITLE
Use database to upload dumpling files to

### PR DESCRIPTION
**Pull request by Szpoti**
**Work In Progress**

### CODE REVIEW

**Comment by molnard** [comment]
Don't forget to remove the FTP script.

**Comment by molnard** [comment]
Say when...

**Comment by Szpoti** [comment]
> Say when...

I will :D Soon...
Final testing is happening

File: Dumplings/Displaying/Connect.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using System.Collections.Generic;
+using System.Text;
+using MySql.Data.MySqlClient;
+
+
+namespace Dumplings.Cli
+{
+    internal class Connect
+    {
+        public const string server = "178.238.222.15";
```
Comment (molnard) [inline Dumplings/Displaying/Connect.cs]: https://learn.microsoft.com/en-us/troubleshoot/developer/visualstudio/csharp/language-compilers/store-custom-information-config-file

File: Dumplings/Displaying/Connect.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            builder.Password = psw;
+            builder.Database = db;
+            string conn = builder.ToString();
+            try
+            {
+                return new MySqlConnection(conn);
+            }
+            catch (MySqlException err)
+            {
+                Console.WriteLine("Nem sikerült a kapcsolat" + err.Message);
```
Comment (molnard) [inline Dumplings/Displaying/Connect.cs]: Ángol!

File: Dumplings/Displaying/Connect.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                return default(MySqlConnection);
```
Comment (molnard) [inline Dumplings/Displaying/Connect.cs]: return null?

File: Dumplings/Displaying/Connect.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            MySqlConnectionStringBuilder builder = new MySqlConnectionStringBuilder();
+            builder.Server = server;
+            builder.UserID = user;
+            builder.Password = psw;
+            builder.Database = db;
+            string conn = builder.ToString();
+            try
+            {
+                return new MySqlConnection(conn);
```
Comment (molnard) [inline Dumplings/Displaying/Connect.cs]: Disposable?

File: Dumplings/Constants.cs [hunk 5]
```diff
@@ -28,6 +28,7 @@ public static class Constants
         public const ulong FirstSamouraiBlock = 570000;
 
         public const ulong FirstWasabiNoCoordAddressBlock = 610000;
+        public static uint256 FirstWW1CoinJoinAfterNoCoordAddress = uint256.Parse("3e044598c2554f8b4479644aa3fc730d5d72ea5d2c5a3c002538268a2ea2f0fe");
```
Comment (molnard) [inline Dumplings/Constants.cs:31]: Add a comment about what this is.

File: Dumplings/Displaying/Connect.cs (hunk 1 above)
Comment (molnard) [reply Dumplings/Displaying/Connect.cs]: Or you can add as argument in CLI as a connection string. Examples here https://www.connectionstrings.com/mysql/

File: Dumplings.Cli/RunAndSendDumplings.sh [hunk 6]
```diff
@@ -3,63 +3,14 @@
...
-FILE8='WasabiPostMixTxs.txt'
-FILE9='WasabiCoinJoins.txt'
-
 # Make sure to have 'Config.txt' in the same file as this script.
 # The following variables need to be imported from the config file:
 
-# HOST = FTP server address,
-# FTPUSER & FTPPASSWD = FTP login credentials,
 # RPCUSER & RPCPASSWD = RPC credentials for Bitcoin Kntos,
-# SCANNERFOLDER = Where to look for the coinjoin stat files to send
```
Comment (molnard) [inline Dumplings.Cli/RunAndSendDumplings.sh:14]: Add more comments on what this file does, how often, etc.

File: Dumplings.Cli/RunAndSendDumplings.sh [hunk 7]
```diff
@@ -3,63 +3,14 @@
...
 
 source Config.txt
 
 echo "Syncronizing blockchain"
```
Comment (molnard) [inline Dumplings.Cli/RunAndSendDumplings.sh]: This is false.

WishYouGIF

File: Dumplings/Stats/Statista.cs [hunk 8]
```diff
@@ -1,87 +1,332 @@
...
+                    cmd.Parameters.AddWithValue("@o", otheri);
+                    cmd.Parameters["@o"].Direction = ParameterDirection.Input;
+                    conn.Open();
+                    int res = cmd.ExecuteNonQuery();
+                    conn.Close();
+                }
+                else
+                {
+                    string sql = $"CALL update{table}(@d,@w,@w2,@s,@o);";
+                    MySqlCommand cmd = new MySqlCommand(sql, conn);
```
Comment (molnard) [inline Dumplings/Stats/Statista.cs]: Disposable?

**Review by molnard — CHANGES_REQUESTED** [review]

File: Dumplings.Cli/GetAndSendWWCoordStats.sh [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+# OUTFOLDER = Folder where Dumplings can save files with desired data.
+# WW1XPUBS & WW2XPUBS = Coordinator Xpubs, separated by coma if more than one.
+
+source Config.txt
+
+echo "Getting Coordinator Stats"
+dotnet run -c Release -- WasabiCoordStats --xpub=$WW1XPUBS --rpcuser=$RPCUSER --rpcpassword=$RPCPASSWD --outfolder=$OUTFOLDER --nowaitonexit
+dotnet run -c Release -- WabiSabiCoordStats --xpub=$WW2XPUBS --rpcuser=$RPCUSER --rpcpassword=$RPCPASSWD --outfolder=$OUTFOLDER --nowaitonexit
+
+# Send files from OUTFOLDER to email address
```
Comment (molnard) [inline Dumplings.Cli/GetAndSendWWCoordStats.sh]: Talk with Kristaps to open a folder for Balint and set outfolder according to that.

**Review by molnard — CHANGES_REQUESTED** [review]

**Comment by molnard** [comment]
Ready to merge?

**Comment by Szpoti** [comment]
I resolve the comments and after that, yes

**Comment by Szpoti** [comment]
Ready

**Review by molnard — APPROVED** [review]
ACK

**Pull request merged**